tests/beta/test_default_tensor.py

```diff
@@ -331,6 +338,298 @@ def test_hermitian_exceptions(self):
+        assert dev._nodes["state"][0].name == "GHZState(0, 1, 2)"
+        assert edges_valid(dev, num_nodes=1)
+
+
+    @pytest.mark.parametrize("tensors,wires,names", [
+        ([np.array([[1., 0.], [0., 1.]]) / np.sqrt(2)], [[0,1]], ["A", "B"]),
+        ([np.array([[1., 0.], [0., 1.]]) / np.sqrt(2)], [[0], [1]], ["A"]),
+        ([np.array([1., 0.]), np.array([1., 1.]) / np.sqrt(2)], [[0]], ["A"]),
+    ])
+    def test_add_initial_state_nodes_exception(self, tensornet_device_2_wires, tensors, wires, names):
+        """Tests that an exception is given if the method _add_initial_state_nodes
+        receives arguments with incompatible lengths"""
+
+        dev = tensornet_device_2_wires
+        dev._clear_network()
+
+        with pytest.raises(ValueError, match="must all be the same length"):
+            dev._add_initial_state_nodes(tensors, wires, names)
+
+
+    def test_add_node(self, tensornet_device_2_wires, tol):
+        """Tests that the _add_node method functions correctly."""
+
+        dev = tensornet_device_2_wires
+
+        assert len(dev._nodes["state"]) == 2
+
+        zero_state = np.array([1., 0])
+        one_qubit_gate = np.array([[0, 1], [1, 0]])
+        two_qubit_gate = np.eye(4)
+        dev._add_node(one_qubit_gate, wires=[0], name="NewNodeX")
+        dev._add_node(one_qubit_gate, wires=[1], name="NewNodeY")
+        dev._add_node(two_qubit_gate, wires=[0, 1], name="NewNodeZ")
+        assert len(dev._nodes["state"]) == 5
+        node_names = [n.name for n in dev._nodes["state"]]
+        print(node_names)
+        assert set(node_names) == set(["ZeroState(0,)",
+                                       "ZeroState(1,)",
+                                       "NewNodeX(0,)",
+                                       "NewNodeY(1,)",
+                                       "NewNodeZ(0, 1)"])
+        tensors = [n.tensor for n in dev._nodes["state"]]
+        assert all([np.allclose(t.data, zero_state, atol=tol, rtol=0) for t in tensors[:2]])
+        assert all([np.allclose(t.data, one_qubit_gate, atol=tol, rtol=0) for t in tensors[2:4]])
+        assert all([np.allclose(t.data, one_qubit_gate, atol=tol, rtol=0) for t in tensors[2:4]])
+        assert np.allclose(tensors[4].data, two_qubit_gate)
+
+        assert "state" in dev._nodes and len(dev._nodes) == 1
+        dev._add_node(one_qubit_gate, wires=[0], key="junk")
+        assert "junk" in dev._nodes and len(dev._nodes) == 2
+
+
+    def test_add_edge(self, tensornet_device_2_wires):
+        """Tests the _add_edge method."""
+
+        dev = tensornet_device_2_wires
+
+        observable = np.array([[0, 1], [1, 0]])
+        obs_nodeA = tensornetwork.Node(observable, name="ObsA")
+        obs_nodeB = tensornetwork.Node(observable, name="ObsB")
+        dev._add_edge(obs_nodeA, 1, dev._nodes["state"][0], 0)
+        dev._add_edge(obs_nodeB, 1, dev._nodes["state"][1], 0)
+
+        # check_connected raises an exception if nodes are not connected
```

Comment:
Is this something that could happen in this specific test? Or maybe it would make a new test case to check for the raised Exception?

Reply:
This is just a note to explain why there is no assertion. i.e., the way that functions is implemented in tensornetwork is it don't return True/False, but just raises an exception. Here a "passing" test is one which doesn't raise an exception. From what I've read, the common-sense way to assert "an exception wasn't raised" is if the test passes

Reply:
Oh got ya! Yes, I agree, makes complete sense. Thank you :)

Reply:
Have updated the name/docstring/comment to explain